client/my-sites/sidebar/sidebar.jsx

```diff
@@ -641,18 +586,41 @@ function mapStateToProps( state ) {
 	const currentUser = getCurrentUser( state );
 	const selectedSiteId = getSelectedSiteId( state );
 	const isSingleSite = !! selectedSiteId || currentUser.site_count === 1;
-	const singleSiteId = selectedSiteId || ( isSingleSite && getPrimarySiteId( state ) ) || null;
-	const selectedSite = getSelectedSite( state );
+	const siteId = selectedSiteId || ( isSingleSite && getPrimarySiteId( state ) ) || null;
+	const site = getSite( state, siteId );
+
+	const isJetpack = isJetpackSite( state, siteId );
+	// FIXME: Fun with Boolean algebra :-)
+	const isSharingEnabledOnJetpackSite = ! (
+		! isJetpackModuleActive( state, siteId, 'publicize' ) &&
+		( ! isJetpackModuleActive( state, siteId, 'sharedaddy' ) || isJetpackMinimumVersion( state, siteId, '3.4-dev' ) )
+	);
+	// FIXME: Turn into dedicated selector
+	const canManagePlugins = !! getSites( state ).some( ( s ) => (
+		( s.capabilities && s.capabilities.manage_options )
+	) );
+	// FIXME: Turn into dedicated selector
+	const hasJetpackSites = getSites( state ).some( s => s.jetpack );
 
 	return {
+		atEnabled: isATEnabled( site ),
+		canManagePlugins,
+		canUserEditThemeOptions: canCurrentUser( state, siteId, 'edit_theme_options' ),
+		canUserListUsers: canCurrentUser( state, siteId, 'list_users' ),
+		canUserManageOptions: canCurrentUser( state, siteId, 'manage_options' ),
+		canUserPublishPosts: canCurrentUser( state, siteId, 'publish_posts' ),
+		canUserViewStats: canCurrentUser( state, siteId, 'view_stats' ),
 		currentUser,
-		atEnabled: isATEnabled( selectedSite ),
 		customizeUrl: getCustomizerUrl( state, selectedSiteId ),
+		hasJetpackSites,
 		isDomainOnly: isDomainOnlySite( state, selectedSiteId ),
-		isJetpack: isJetpackSite( state, selectedSiteId ),
+		isJetpack,
+		isSharingEnabledOnJetpackSite,
 		isSiteAutomatedTransfer: !! isSiteAutomatedTransfer( state, selectedSiteId ),
-		menusUrl: getMenusUrl( state, singleSiteId ),
-		singleSiteId
+		menusUrl: getMenusUrl( state, siteId ),
+		siteId,
+		site,
+		siteSuffix: siteId ? '/' + site.slug : '',
```

Comment:
I actually changed this to `siteId` because of my own paranoia -- to avoid the empty object case. Tho that's very unlikely, I guess 😄 And it'd only add the `/`, so it should be fine. Okay, changing it back.